sklbench/utils/data.py

```diff
@@ -0,0 +1,209 @@
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import os
+import requests
+from typing import List, Callable, Union
+
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+from sklearn.model_selection import train_test_split
+from sklearn.datasets import fetch_openml
+
+try:
+    import kaggle
+    kaggle_is_imported = True
+except (ImportError, OSError, ValueError):
+    kaggle_is_imported = False
+
+from .common import get_bench_case_value
+
+
+def convert_data(data, dformat: str, order: str, dtype: str):
+    if isinstance(data, csr_matrix) and dformat != 'csr_matrix':
+        data = data.toarray()
+    if order == 'F':
+        data = np.asfortranarray(data, dtype=dtype)
+    elif order == 'C':
+        data = np.ascontiguousarray(data, dtype=dtype)
+    else:
+        raise ValueError(f'Unknown data order {order}')
+    if dformat == 'numpy':
+        return data
+    elif dformat == 'pandas':
+        if data.ndim == 1:
+            return pd.Series(data)
+        return pd.DataFrame(data)
+    else:
+        raise ValueError(f'Unknown data format {dformat}')
+
+
+def train_test_split_wrapper(*args, **kwargs):
+    if 'ignore' in kwargs:
+        result = []
+        for arg in args:
+            result += [arg, arg]
+        return result
+    else:
+        return train_test_split(*args, **kwargs)
+
+
+def split_and_transform_data(bench_case, data, data_description):
+    if 'default_split' in data_description:
+        split_kwargs = data_description['default_split'].copy()
+    else:
+        split_kwargs = {
+            'random_state': 42

```

Comment:
For user reproducibility outside of the benchmarks, this needs to be better exposed to the user. It would be best to include these parameters in the reports and the json files (size of split and random_state that were passed).